### PR TITLE
penalty_tx by block conflict, some refactor

### PR DESCRIPTION
**Pull request by SteelStrider**

### CODE REVIEW

File: node/node.py [hunk 1]
```diff
@@ -335,15 +341,10 @@ def form_private_key_reveal_transaction(self):
...
-        # TODO can be deprecated on get block by ancestor complete logic
-        for conflict in conflicts:
-            block = self.dag.blocks_by_hash[conflict]
-            self.network.broadcast_block(self.node_id, block.pack())
         self.logger.info("Forming transaction with conflicting blocks")
-        self.logger.info(conflict.hex())
         node_private = self.block_signer.private_key
         tx = TransactionFactory.create_penalty_transaction(conflicts, node_private)
-        return tx
+        return [tx]
```
Comment (konchunas) [inline node/node.py]: Why return array here? If you need to return array, why left singular name?

File: node/node.py [hunk 2]
```diff
@@ -194,16 +195,20 @@ def sign_block(self, current_block_number):
...
         chosen_top = self.dag.get_longest_chain_top(tops)
         conflicting_tops = [top for top in tops if top != chosen_top]
-        
+
         current_top_blocks = [chosen_top] + conflicting_tops  # first link in dag is not considered conflict, the rest is.
-        
+
+        # PROVIDE PENALTY TRANSACTION FOR CONFLICT TOPS
+        if len(conflicting_tops) > 0:
+            system_txs += self.form_penalize_violators_transaction(conflicting_tops)
```
Comment (konchunas) [inline node/node.py]: Why not do an .append() here?

File: node/node.py [hunk 3]
```diff
@@ -540,6 +541,20 @@ def insert_verified_block(self, signed_block, allowed_pubkey):
         block_number = self.epoch.get_block_number_from_timestamp(block.timestamp)
         epoch_number = Epoch.get_epoch_number(block_number)
 
+        # GET_CONFLICT_HASHES
+        system_txs = block.system_txs
+        conflict_block_hashes = []
+        for system_tx in system_txs:
+            if isinstance(system_tx, PenaltyTransaction):
+                # penalty transaction for block conflicts
+                conflict_block_hashes = system_tx.conflicts
```
Comment (konchunas) [inline node/node.py]: Did you mean to use += here?

File: tests/helper_test.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+from chain.params import ROUND_DURATION
+from crypto.private import Private
+from node.behaviour import Behaviour
+from node.block_signers import BlockSigner
+from node.node import Node
+from tools.time import Time
+
+
+class HelperTest(unittest.TestCase):
```
Comment (konchunas) [inline tests/helper_test.py]: This should be named "TestHelper", because it is not testing helper. It helps to test.

File: node/node.py (hunk 1 above)
Comment (SteelStrider) [reply node/node.py]: corrected

File: node/node.py (hunk 2 above)
Comment (SteelStrider) [reply node/node.py]: corrected

File: node/node.py (hunk 3 above)
Comment (SteelStrider) [reply node/node.py]: Agree, corrected

File: tests/helper_test.py (hunk 4 above)
Comment (SteelStrider) [reply tests/helper_test.py]: Agree

**Pull request closed without merging**